nautobot/dcim/filters/__init__.py

```diff
@@ -1925,3 +1809,180 @@ def _subtree(self, queryset, name, value):
         """FilterSet method for getting Groups that are or are descended from a given ControllerManagedDeviceGroup(s)."""
         params = self.generate_query__subtree(value)
         return queryset.filter(params)
+
+
+class ModuleFilterSet(
+    DeviceModuleCommonFiltersMixin,
+    LocatableModelFilterSetMixin,
+    RoleModelFilterSetMixin,
+    StatusModelFilterSetMixin,
+    TenancyModelFilterSetMixin,
+    NautobotFilterSet,
+):
+    q = SearchFilter(
+        filter_predicates={
+            "asset_tag": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "serial": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "module_type__manufacturer__name": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+            "module_type__model": {
+                "lookup_expr": "icontains",
+                "preprocessor": str.strip,
+            },
+        }
+    )
+    is_installed = RelatedMembershipBooleanFilter(
+        field_name="parent_module_bay",
+        label="Is installed in a module bay",
+    )
+    manufacturer = NaturalKeyOrPKMultipleChoiceFilter(
+        field_name="module_type__manufacturer",
+        queryset=Manufacturer.objects.all(),
+        to_field_name="name",
+        label="Manufacturer (name or ID)",
+    )
+    module_type = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=ModuleType.objects.all(),
+        to_field_name="model",
+        label="Module type (model or ID)",
+    )
+    parent_module_bay = django_filters.ModelMultipleChoiceFilter(
+        queryset=ModuleBay.objects.all(),
+        label="Parent Module Bay",
+    )
+
+    class Meta:
+        model = Module
+        fields = "__all__"
```

Comment:
Something I noticed yesterday is that when we're using `__all__` on our new FilterSets, this includes a `_custom_field_data` filter, which the GraphQL schema generation complains about because GraphQL filters can't start with an underscore. Not blocking for this PR in any way, but we should fix that, possibly by enhancing `NautobotFilterSet` to auto-exclude `_custom_field_data` as a filter?

Reply:
> #5703.

Thanks!